src/components/CardWrapper.tsx

```diff
@@ -0,0 +1,27 @@
+import React from "react";
+import { useLocation } from "react-router-dom";
+
+import BingoCard from "./BingoCard.tsx";
+
+import cards from "../assets/cards.json";
+
+function CardWrapper() {
+  const useQuery = () => new URLSearchParams(useLocation().search);
+  const query = useQuery();
+  const cardId = query.get("card");
+
+  if (!cardId) return <div>Error reading card id!</div>;
+
+  const cardIndex = parseInt(cardId) % cards.length; // Wrap around for invalid card IDs
```

Comment:
❓ Should we prompt the user to select a card if the index is out of bounds? Worried we might select a default for the user without them being aware.

Reply:
good idea